### PR TITLE
+f

### PR DESCRIPTION
**Pull request by zach1502**
My condolences Injabie.

#125

### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature
- [x] F

### Description of the changes
+f

### CODE REVIEW

**Comment by zach1502** [comment]
F

**Comment by zach1502** [comment]
F

**Comment by zach1502** [comment]
hiteler

File: cogs/Respects.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class Respects:
+    '''Press "F" to pay respects'''
+    def __init__(self, bot):
+        self.bot = bot
+    @commands.command(name="f", pass_context=True) # pylint-disable=too-few-public-methods
+    async def mycom(self, context):
+        '''Press "F" to pay respects'''
+        await bot.say(context.message.author + "hAs pAiD rEsPeCtS")
+
```
Comment (Injabie3) [inline cogs/Respects.py:36]: Where is the cog set up?

File: cogs/Respects.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'''Press "F" to pay respects'''
+import asyncio
+import discord
+
+class Respects:
+    '''Press "F" to pay respects'''
+    def __init__(self, bot):
+        self.bot = bot
+    @commands.command(name="f", pass_context=True) # pylint-disable=too-few-public-methods
+    async def mycom(self, context):
```
Comment (Injabie3) [inline cogs/Respects.py]: This does not address #125.

**Review by Injabie3 — CHANGES_REQUESTED** [review]

File: cogs/Respects.py (hunk 2 above)
Comment (zach1502) [reply cogs/Respects.py]: dont worry, i got this

File: cogs/Respects.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+class Respects:
+    '''Press "F" to pay respects.'''
+    def __init__(self, bot):
+        self.bot = bot
+    @commands.command(name="f", pass_context=True) # pylint-disable=too-few-public-methods
+    async def mycom(self, ctx):
+        '''Press "F" to pay respects.'''
+        who_F = []
+        who_F.append(ctx + ", ")
```
Comment (Injabie3) [inline cogs/Respects.py]: Did you mean`ctx.message.author`?

File: cogs/Respects.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class Respects:
+    '''Press "F" to pay respects.'''
+    def __init__(self, bot):
+        self.bot = bot
+    @commands.command(name="f", pass_context=True) # pylint-disable=too-few-public-methods
+    async def mycom(self, ctx):
+        '''Press "F" to pay respects.'''
+        who_F = []
```
Comment (Injabie3) [inline cogs/Respects.py]: This is the right idea: we want a list of people who have reacted.  The next step is to figure out how to store this across multiple invocations of this command.  Off the top of my head, we'll probably want a helper function to do some kind of a lookup to see when the latest invocation was, and if it exceeds, then we want a new "has paid respects", else append to the old one, delete the pre-existing "respects", and put a new one.

File: cogs/Respects.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        who_F.append(ctx + ", ")
+        hearts = ["💚", "💙", "💜", "💘", "🧡", "♥️", "💓", "💖", "💕", "💛", "💗", "💞", "💝", "❤️", "🖤"]
+        await bot.say(who_F + "has paid their respects " + random.choice(hearts))
+
+        await asyncio.sleep(10)
```
Comment (Injabie3) [inline cogs/Respects.py]: This sleep seems unnecessary.  We don't need to wait 10 seconds for this function, multiple users can still invoke this function from a channel anyways.

File: cogs/Respects.py (hunk 4 above)
Comment (Injabie3) [reply cogs/Respects.py]: Right now, this is only local to each invocation of the command, we want this to exist over multiple invocations.

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Hey zach, please see some comments below.

**Comment by Injabie3** [comment]
As per our discussion, you could do something like the following:
1. Someone invokes command.
2. See where it comes from (e.g. from what server and channel)
3. Do a lookup to see when the last invocation of that was (this you have to make, e.g. a dictionary with perhaps a nested key with server and channel IDs).  Two cases: there was a previous invocation or there wasn't.
4. If there was no invocation, just send the message with who has paid respects AND save the message ID, and save who has paid respects, the message ID, and the invocation time to the lookup.
5. If there was, check the time, and see if it's within a threshold (you determine).  If the time has passed, create a new one.  Else just delete the previous message and make a new one in the same manner as above.

This is just a rough guide from the idea I was thinking of on the back of some rough paper, perhaps you may find a better approach.

**Comment by Injabie3** [comment]
Don't see anything happening lately with this PR, so I will try something out.

**Comment by Injabie3** [comment]
Closing for #154 merged in at commit 7a60cec2e3aa919ae966dc49f9daf06cd56fabf3.

**Pull request closed without merging**